src/abstractdataframe/abstractdataframe.jl

```diff
@@ -763,7 +763,7 @@ Base.hcat(df1::AbstractDataFrame, df2::AbstractDataFrame, dfn::AbstractDataFrame
 
 @generated function promote_col_type(cols::AbstractVector...)
     T = mapreduce(x -> Missings.T(eltype(x)), promote_type, cols)
-    if T <: CategoricalValue
+    if CategoricalArrays.iscatvalue(T)
         T = T.parameters[1]
```

Comment:
Sorry, I haven't looked at the whole function when I gave the last comment.
This line would give incorrect result for e.g. `CategoricalString{UInt8}`.
The proper way should be `T = CategoricalArrays.leveltype(T)`

Reply:
Good catch. Note that REQUIRE should also be raised to the latest CategoricalArrays version.

Reply:
@nalimilan It seems like REQUIRE already requires the latest CategoricalArrays (version 0.3.3)?